### ISSUE TITLE
Undocumented configuration change?

### ISSUE THREAD

**Issue by pvangent** [gpc/grails-joda-time#28]
I am busy with a (long overdue) upgrade of a Grails2 app to Grails3. Part of the upgrade is maintaining the Joda Time functionality. This has likely succeeded, but with a configuration that is not documented. I am not sure if the plugin's workings have changed, or the documentation is incorrect, but I think it's worth noting.

Some facts about the app:
- Grails Version: 3.3.6
- Groovy Version: 2.4.15
- JVM Version: 1.8.0_161
- Hibernate: 5.1.5

Joda (related) dependencies in build.gradle:
```
compile 'org.grails.plugins:joda-time:2.1.0'
compile "org.jadira.usertype:usertype.jodatime:1.9"
```

In the Grails2 app the Joda <> Hibernate mapping was in a ```grails.gorm.default.mapping``` entry in Config.groovy. I moved this to application.groovy, which is the correct place according to this plugin's documentation (and install-mapping script). However, in the Grails3 app ```grails.gorm.default.mapping``` does not work. What does work, is using the ```grails.gorm.default.constraints``` entry. Like so:
```
grails.gorm.default.constraints = {
    "user-type" type: org.jadira.usertype.dateandtime.joda.PersistentDateTime, class: org.joda.time.DateTime
    "user-type" type: org.jadira.usertype.dateandtime.joda.PersistentDateTimeZoneAsString, class: org.joda.time.DateTimeZone
    "user-type" type: org.jadira.usertype.dateandtime.joda.PersistentDurationAsString, class: org.joda.time.Duration
    "user-type" type: org.jadira.usertype.dateandtime.joda.PersistentInstantAsMillisLong, class: org.joda.time.Instant
    "user-type" type: org.jadira.usertype.dateandtime.joda.PersistentInterval, class: org.joda.time.Interval
    "user-type" type: org.jadira.usertype.dateandtime.joda.PersistentLocalDate, class: org.joda.time.LocalDate
    "user-type" type: org.jadira.usertype.dateandtime.joda.PersistentLocalDateTime, class: org.joda.time.LocalDateTime
    "user-type" type: org.jadira.usertype.dateandtime.joda.PersistentLocalTime, class: org.joda.time.LocalTime
    "user-type" type: org.jadira.usertype.dateandtime.joda.PersistentPeriodAsString, class: org.joda.time.Period
    "user-type" type: org.jadira.usertype.dateandtime.joda.PersistentYears, class: org.joda.time.Years
}
```

(To be clear: it works fine in the app, but it seems undocumented - at least for this version of the plugin?)

**Comment by graemerocher** [comment]
Seems worthy of a pull request to the docs

**Comment by pvangent** [comment]
Right. But doesn't it also require an update of the joda-time/grails-app/commands/grails/plugins/jodatime/InstallJodaTimeGormMappingsCommand.groovy script?

**Comment by pvangent** [comment]
The new config ```grails.gorm.default.constraints``` worked for a while, until I cleaned the application. After that it would not startup again, with a fatal Hibernate error configuring the datasource. I found some references on the web regarding dependencies between Jadira en Hibernate5.x, and it looks like you should use ```org.jadira.usertype:usertype.core:5.0.0.GA``` instead of ```org.jadira.usertype:usertype.jodatime:1.9```. For me it works anyway...

If this is true, that would be another change to the docs I guess?
